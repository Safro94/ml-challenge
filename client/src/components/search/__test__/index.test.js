import React from 'react';

import { render } from '@testing-library/react';
import { screen } from '@testing-library/dom';
import userEvent from '@testing-library/user-event';

import Search from '../';

describe('Search', () => {
	it('should render OK', () => {
		//Act
		const { container } = render(<Search />);

		//Assert
		expect(container).toMatchSnapshot();
	});

	it('should call onSubmit', () => {
		//Arrange
		const onSubmit = jest.fn();
		const term = 'test';

		//Act
		render(<Search onSubmit={onSubmit} />);
		const button = screen.getByRole('button');

		//Assert
		expect(button).toBeDisabled();
		userEvent.type(screen.getByPlaceholderText(/Nunca dejes de buscar/i), term);

		expect(button).not.toBeDisabled();
		userEvent.click(button);

		expect(onSubmit).toHaveBeenCalledTimes(1);
		expect(onSubmit).toHaveBeenCalledWith(term);
	});
});
