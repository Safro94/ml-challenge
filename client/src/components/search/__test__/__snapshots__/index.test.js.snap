// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search should render OK 1`] = `
<div>
  <div>
    <form
      class="form"
    >
      <input
        class="input"
        placeholder="Nunca dejes de buscar"
        type="text"
        value=""
      />
      <button
        class="button button"
        disabled=""
        type="submit"
      >
        <img
          alt="Buscar"
          src="/assets/ic_Search.png"
        />
      </button>
    </form>
  </div>
</div>
`;
